modules/pxp-module-puppet

```diff
@@ -26,14 +26,30 @@ end
 
 DEFAULT_EXITCODE = -1
 
-# This method exists to facilitate testing
-def last_child_exit_status
-  return $?
-end
+env_fix_up = nil
+
+def get_env_fix_up
+  # Prepare an environment fix-up to make up for its cleansing performed
+  # by both pxp-agent and the Puppet::Util::Execution.execute function.
```

Comment:
If you want to do so with POSIX's daemonize(), you need some extra code. We don't do that; forked processes inherit the parent's env.
Besides, if I log {{LOG_INFO("daemonization complete; HOME = {1}", std::getenv("HOME"));}} just after daemonize(), I get this when I run pxp-agent as `alessandro`:
`
2016-05-16 14:16:10.534428 INFO  puppetlabs.pxp_agent.main:57 - daemonization complete; HOME = /Users/alessandro
`

Also, pxp-agent should not be changing the environment when it calls leatherman's execute(): https://github.com/puppetlabs/pxp-agent/blob/master/lib/src/external_module.cc#L360

We should look into that if it's really the case.